File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    let count = 0;
    let camera = -30001;
    routes.sort((a, b) => a[1] - b[1]);
    
    for (const [enter, out] of routes) {
        if (camera < enter) {
            count++;
            camera = out;
        }
    }
    return count;
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda x: (x[1], x[0]))
    
    loc = -float('inf') # -inf -> 15
    for s, e in routes: # -20, -15 -> -18, -13 -> -14, -5 -> -5, -3
        if loc < s:
            answer += 1
            loc = e
        
    return answer